# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../strace/01-linux_sigcontext.patch.mips
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Based on strace patch from OpenWRT:
https://dev.openwrt.org/browser/packages/utils/strace/patches/01-linux_sigcontext.patch?rev=

--- strace-4.5.14/system.c	2006-01-12 22:21:06.000000000 +0100
+++ strace-4.5.14.new/system.c	2006-12-10 15:14:20.000000000 +0100
@@ -78,6 +78,7 @@
 #endif
 
 #include <linux/sysctl.h>
+#include <linux/version.h>
 
 static const struct xlat mount_flags[] = {
 	{ MS_MGC_VAL,	"MS_MGC_VAL"	},
@@ -1578,7 +1579,9 @@
 	{ CTL_KERN, "CTL_KERN" },
 	{ CTL_VM, "CTL_VM" },
 	{ CTL_NET, "CTL_NET" },
-	{ CTL_PROC, "CTL_PROC" },
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,18)
+        { CTL_PROC, "CTL_PROC" },
+#endif
 	{ CTL_FS, "CTL_FS" },
 	{ CTL_DEBUG, "CTL_DEBUG" },
 	{ CTL_DEV, "CTL_DEV" },
--- strace-4.5.15/signal.c.orig	2007-01-11 23:08:38.000000000 +0100
+++ strace-4.5.15/signal.c	2007-01-25 12:00:54.000000000 +0100
@@ -66,7 +66,7 @@
 
 
 #ifdef LINUX
-
+#include <linux/version.h>
 #ifdef IA64
 # include <asm/ptrace_offsets.h>
 #endif /* !IA64 */
@@ -1435,12 +1435,16 @@
 		if (umove(tcp, sp, &sc) < 0)
 		  	return 0;
 		tcp->u_arg[0] = 1;
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,0)
 		tcp->u_arg[1] = sc.sc_sigset;
+#else
+		tcp->u_arg[1] = sc.sc_hi2;
+#endif
 	} else {
 	  	tcp->u_rval = tcp->u_error = 0;
 		if(tcp->u_arg[0] == 0)
 		  	return 0;
-		tcp->auxstr = sprintsigmask("mask now ", tcp->u_arg[1]);
+		tcp->auxstr = sprintsigmask("mask now ", tcp->u_arg[1], 0);
 		return RVAL_NONE | RVAL_STR;
 	}
 	return 0;
